library/src/components/List/Content/Shared/DownloadPdf.tsx

```diff
@@ -0,0 +1,59 @@
+import { useStoreActions, useStoreState } from 'store';
+import { StyledPdfIcon } from './DownloadPdf.styles';
+
+export interface DownloadPdfProps {
+  row: Record<string, any>;
+}
+
+export default function DownloadPdf(props: DownloadPdfProps): JSX.Element {
+  const { row } = props;
+
+  const entities = useStoreState((state) => {
+    return state.entities.entities;
+  });
+
+  const apiDownload = useStoreActions((actions) => {
+    return actions.api.download;
+  });
+
+  const truncate = (string: string, n: number): string => {
+    return string.length > n ? string.substr(0, n - 1) + '...' : string;
+  };
+
+  const downloadPdf = () => {
+    apiDownload({
+      path: `${entities.Invoice.path}/${row.id}/pdf`,
+      params: {},
+      headers: {
+        accept: 'application/octet-stream',
+      },
+      successCallback: async (data, headers) => {
+        const fileName = (headers['content-disposition'] || '')
+          .split('filename=')
+          .pop();
+
+        const blobUrl = URL.createObjectURL(data as Blob);
+        const link = document.createElement('a');
+        link.href = blobUrl;
+        link.download = fileName || 'invoice.pdf';
+
+        document.body.appendChild(link);
+        link.dispatchEvent(
+          new MouseEvent('click', {
+            bubbles: true,
+            cancelable: true,
+            view: window,
+          })
+        );
+        document.body.removeChild(link);
+      },
+    });
+  };
+
+  return (
+    <>
+      <StyledPdfIcon onClick={downloadPdf} />
+      <span>{truncate(row.pdf.baseName, 4)}</span>
+    </>
+  );
+}
```

Comment:
This can be applied for any downloadable file, besides, you can perform this in its own List Content only with a few lines and you can't obtain  the entities like this, this is a library, this can be used by any entity, so remove this file and reorder the logic please